### ISSUE TITLE
[notification] Auto-dismiss

### ISSUE THREAD

**Issue by svenefftinge** [theia-ide/theia#1081]
There are many cases where it would be preferable to dismiss a notification automatically after some time. There should be an option to specify the timeout when no actions are provided.

**Comment by marcdumais-work** [comment]
Regarding the auto-dismiss being optional:

One reason I can see not to automatically dismiss some notifications is that they may contain useful actions. e.g. when the C/C++ LS is not found, there is a button that opens a clangd web page with setup info. If the user turn their attention to something else for a few moment, the notification might be gone. 

One alternative option would be to auto-dismiss based on the type of notification: auto-dismiss log, info, warn but not error?

**Comment by svenefftinge** [comment]
> There should be an option to specify the timeout when no actions are provided.

@marcdumais-work I think that would solve the situation you describe. It is a property that is set if there are no actions included. The time should come from a preference.

**Comment by lmcbout** [comment]
The existence of actions is the first criteria. The notification timeout should depends on actions. When no actions are requiring using intervention, we set a timeout to auto-dismiss the notification. It could be set when calling the MessageService like ( this.messageService.error(`Error launching task`, { timeout: 3000 }); ) or could be set in preferences (In <workspace>/.theia/settings.json like 
{
    // Time in millisec to auto-dismiss a notification
	"notification.timeout": 2000
}
If we set the timeout to "0", then the auto-dismiss will not apply and the user will need to "Close" the notification.
We calculate the timeout like this?
const timeout = (message.actions)
            ? undefined
            : (message.timeout !== undefined ? message.timeout : this.preferences['notification.timeout']);
